inferno/dataset.py

```diff
@@ -168,3 +185,140 @@ def __getitem__(self, i):
+      Possible inputs for cv are:
+
+        - None, to use the default 3-fold cross validation,
+        - integer, to specify the number of folds in a `(Stratified)KFold`,
+        - float, to represent the proportion of the dataset to include
+          in the test split.
+        - An object to be used as a cross-validation generator.
+        - An iterable yielding train, test splits.
+
+    classifier : bool (default=False)
+      For integer/None inputs to cv, if the estimator is a classifier
+      (e.g. `NeuralNetClassifier`) and `y` is either binary or
+      multiclass, sklearn's StratifiedKFold is used. In all other
+      cases, KFold is used.
+
+      """
+    def __init__(self, cv=5, classifier=False):
+        self.classifier = classifier
+
+        if isinstance(cv, Number) and (cv <= 0):
+            raise ValueError("Numbers less than 0 are not allowed for cv "
+                             "but CVSplit got {}".format(cv))
+        self.cv = cv
+
+    def regular_cv(self, X, y, cv):
+        """Use the normal `.split` interface for data types are
+        supported by sklearn.
+
+        """
+        idx_train, idx_valid = next(iter(cv.split(X, y)))
+        X_train = safe_indexing(X, idx_train)
+        X_valid = safe_indexing(X, idx_valid)
+        y_train = None if y is None else safe_indexing(y, idx_train)
+        y_valid = None if y is None else safe_indexing(y, idx_valid)
+        return X_train, X_valid, y_train, y_valid
+
+    def special_cv(self, X, y, cv, stratified=False):
+        """For data types not directly supported by sklearn, use
+        custom split function.
+
+        """
+        dataset = Dataset(X, y)
+        num_samples = len(dataset)
+        args = (np.arange(num_samples),)
+        if stratified:
+            y_arr = to_numpy(y)
+            args = args + (y_arr,)
+        idx_train, idx_valid = next(iter(cv.split(*args)))
+
+        X_train, y_train = dataset[idx_train]
+        X_valid, y_valid = dataset[idx_valid]
+        return X_train, X_valid, y_train, y_valid
+
+    def _check_cv_float(self, y):
+        stratified = False
+        if y is None:
+            cv = check_cv(
+                ShuffleSplit(test_size=self.cv),
+                classifier=self.classifier,
+            )
+            return cv, stratified
+
+        cv_cls = StratifiedShuffleSplit if self.classifier else ShuffleSplit
+        try:
+            cv = check_cv(
+                cv_cls(test_size=self.cv),
+                y=y,
+                classifier=self.classifier,
+            )
+            stratified = True
+        except ValueError:
+            cv = check_cv(
+                ShuffleSplit(test_size=self.cv),
+                classifier=self.classifier,
+            )
+        return cv, stratified
+
+    def _check_cv(self, y):
+        try:
+            # for stratified split, y must be a numpy array
+            y_arr = to_numpy(y)
+        except AttributeError:
+            y_arr = y
```

Comment:
Why is it a good idea to take the valuethat just caused an attribute error? Should be documented.